ext/KrylovKitChainRulesCoreExt/svdsolve.jl

```diff
@@ -0,0 +1,289 @@
+        else
+            Δs = real(Δσ)
+        end
+        Δu = add(Δu, u, -udΔu)
+        Δv = add(Δv, v, -vdΔv)
+        b = (Δu, Δv)
+        (x, y), reverse_info = let σ = σ, u = u, v = v
+            linsolve(b, zerovector(b), alg_rrule) do z
+                x, y = z
+                x′ = VectorInterface.add!!(apply_normal(f, y), x, σ, -1)
+                y′ = VectorInterface.add!!(apply_adjoint(f, x), y, σ, -1)
+                x′ = VectorInterface.add!!(x′, u, -inner(u, x′))
+                y′ = VectorInterface.add!!(y′, v, -inner(v, y′))
+                return (x′, y′)
+            end
+        end
+        if info.converged >= i && reverse_info.converged == 0 && alg_rrule.verbosity >= 0
+            @warn "`svdsolve` cotangent linear problem ($i) did not converge, whereas the primal eigenvalue problem did: normres = $(reverse_info.normres)"
+        end
+        x = VectorInterface.add!!(x, u, Δs / 2)
+        y = VectorInterface.add!!(y, v, conj(Δs) / 2)
+        xs[i] = x
+        ys[i] = y
+    end
+    return xs, ys
+end
+function compute_svdsolve_pullback_data(Δvals, Δlvecs, Δrvecs, vals, lvecs, rvecs, info, f,
+                                        which, alg_primal, alg_rrule::Arnoldi)
+    @assert which == :LR
+    T = scalartype(lvecs[1])
+    n = length(Δvals)
+    UdΔU = zeros(T, n, n)
+    VdΔV = zeros(T, n, n)
+    for j in 1:n
+        for i in 1:n
+            if !(Δlvecs[j] isa AbstractZero)
+                UdΔU[i, j] = inner(lvecs[i], Δlvecs[j])
+            end
+            if !(Δrvecs[j] isa AbstractZero)
+                VdΔV[i, j] = inner(rvecs[i], Δrvecs[j])
+            end
+        end
+    end
+    aUdΔU = rmul!(UdΔU - UdΔU', 1 / 2)
+    aVdΔV = rmul!(VdΔV - VdΔV', 1 / 2)
+
+    tol = alg_primal.tol
+    mask = abs.(vals' .- vals) .< tol
+    gaugepart = view(aUdΔU, mask) + view(aVdΔV, mask)
+    gaugeerr = norm(gaugepart, Inf)
+    if gaugeerr > alg_primal.tol && alg_rrule.verbosity >= 1
+        @warn "`svdsolve` cotangents for singular vectors are sensitive to gauge choice: (|gauge| = $gaugeerr)"
+    end
+    UdΔAV = (aUdΔU .+ aVdΔV) .* safe_inv.(vals' .- vals, tol) .+
+            (aUdΔU .- aVdΔV) .* safe_inv.(vals' .+ vals, tol)
+    if !(Δvals isa ZeroTangent)
+        UdΔAV[diagind(UdΔAV)] .+= real.(Δvals)
+    end
+
+    xs = similar(lvecs, n)
+    ys = similar(rvecs, n)
+    for i in 1:n
+        x = scale(lvecs[1], UdΔAV[1, i] / 2)
+        y = scale(rvecs[1], conj(UdΔAV[i, 1]) / 2)
+        for j in 2:n
+            x = VectorInterface.add!!(x, lvecs[j], UdΔAV[j, i] / 2)
+            y = VectorInterface.add!!(y, rvecs[j], conj(UdΔAV[i, j]) / 2)
+        end
+        xs[i] = x
+        ys[i] = y
+    end
+
+    sylvesterargx = similar(lvecs)
+    for i in 1:n
+        x = zerovector(lvecs[1])
+        if !(Δlvecs[i] isa AbstractZero)
+            x = VectorInterface.add!!(x, Δlvecs[i], +1)
+            for j in 1:n
+                y = VectorInterface.add!!(x, lvecs[j], -UdΔU[j, i])
```

Comment:
```suggestion
                x = VectorInterface.add!!(x, lvecs[j], -UdΔU[j, i])
```
I think this is a small typo?